kafka-client/src/main/java/dev/responsive/kafka/api/KeyValueStoreOptions.java

```diff
@@ -0,0 +1,92 @@
+package dev.responsive.kafka.api;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+/**
+ * Configuration options for building a Responsive {@link KeyValueStore}. Use {@link #named(String)}
+ * (or {@link #namedGlobal(String)} if this is a global store) to get a new object and pass that in
+ * to one of the {@link ResponsiveStoreFactory} APIs, or apply any of the additional optional
+ * configs via the {@code #withXXX} methods below.
+ */
+public class KeyValueStoreOptions<K, V> {
+
+  // required configs
+  private final String name;
+  private final boolean isGlobal;
```

Comment:
I suspect you're creating this because with the current API we need to know whether or not the store is global up front. I don't think we actually need this though, we can get rid of `ResponsiveGlobalStore` and have only one `ResponsiveStore` that has two different internal implementations and we check whether the `StateStoreContext` that is passed in is `instanceof GlobalProcessorContextImpl` (or `taskId().partition() == -1` if we don't want to use instanceof) to initialize the "global" version of it

The benefit is that we can keep the `ResponsiveStoreFactory` API identical to `Stores` and it's one less thing for the user to worry about.

Reply:
I want to push back a _bit_ on replicating the existing `Stores` exactly, but when it comes to the global store thing I agree 100%. It would be way too easy for someone to miss this and pass in a non-global store by accident, and it's clunky. So if you found a way to figure out which kind of store it is implicitly, I'm all for it!

Reply:
I do want to make sure the basic API is done ASAP so I'll just remove the `isGlobal` stuff altogether for now. Would you mind filing a ticket for refactoring the global store implementation based on this idea to check the type during `init`?

Reply:
Yeah I can take this Action Item.